### PR TITLE
[Collections] Support 'v' prefix in version

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
Package versions with 'v' prefix (e.g., `v1.0.0`) in a collection are dropped because they are not semver, but SwiftPM actually supports checking out git tags that have 'v' prefix.

Modifications:
- Move `TSCUtility.Verson.init(tag:)` to `Basics` for sharing
- Use `Version(tag:)`, which strips away any 'v' prefix, instead of `Version(string:)`

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/Basics/Version+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    ///
+    /// - Parameter tag: A version string possibly prepended with "v".
+    public init?(tag: String) {
+        if tag.first == "v" {
+            self.init(string: String(tag.dropFirst()))
+        } else {
+            self.init(string: tag)
+        }
+    }
+}
```
Comment (tomerd) [inline Sources/Basics/Version+Extensions.swift:24]: add to cmake?

**Review by tomerd — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/Basics/Version+Extensions.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/Basics/Version+Extensions.swift:24]: Yep https://github.com/apple/swift-package-manager/pull/3495/commits/2fcdfd5655656617c417d9f0d2b5f33b88ae8eec

**Comment by yim-lee** [comment]
Hmm test failure doesn't seem to be related to this PR but got the same failure twice already:

```
21:29:49 Test Case 'BuildToolTests.testBuildCompleteMessage' started at 2021-05-13 04:29:06.495
21:29:49 
/home/buildnode/jenkins/workspace/swift-package-manager-self-hosted-Linux-smoke-test/branch-main/swiftpm/Tests/CommandsTests/BuildToolTests.swift:256: error: BuildToolTests.testBuildCompleteMessage : XCTAssertTrue failed - [1/4] Compiling Bar Bar.swift
21:29:49 [3/6] Merging module Bar
21:29:49 [5/7] Wrapping AST for Bar for debugging
21:29:49 [6/9] Compiling Foo Foo.swift
21:29:49 [7/9] Compiling Foo main.swift
21:29:49 [9/11] Merging module Foo
21:29:49 [11/12] Wrapping AST for Foo for debugging
21:29:49 [12/12] Linking Foo
21:29:49 [12/12] Build complete!
21:29:49 
21:29:49 Test Case 'BuildToolTests.testBuildCompleteMessage' failed (5.24 seconds)
```

@swift-ci please smoke test Linux

**Comment by yim-lee** [comment]
CI failure reproducible with `swiftlang/swift:nightly-focal`. The assertion in `BuildToolTests.testBuildCompleteMessage` expects `[8/8] Build complete!` but looks like it's become `[12/12] Build complete!`.

**Comment by yim-lee** [comment]
@swift-ci please smoke test self hosted Linux

**Comment by yim-lee** [comment]
@swift-ci please smoke test self hosted

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Pull request merged**